samples/js/fb/find-photos/src/main.js

```diff
@@ -111,3 +128,24 @@ async function getSizes(input): Promise<Map<Struct, string>> {
   });
   return result;
 }
+
+async function getFaces(photo): Promise<Set<Struct>> {
+  // TODO: Facebook only gives us the centerpoint of each face, not the
+  // bounding box. Ideally, we'd use face detection on the image to get the
+  // real bounding box, but for now, we just guess that faces are typically
+  // about 1/3 the width/height of image.
+  //
+  // This fails badly in lots of cases though, so we should fix asap.
+  const mw = 0.33;
+  const mh = 0.33;
+  let result = Promise.resolve(new Set());
+  await photo.tags.data.forEach(v => {
+    const x = Math.max(0, v.x / 100 - mw / 2);
+    const y = Math.max(0, v.y / 100 - mh / 2);
+    const w = Math.min(mw, 1 - x);
+    const h = Math.min(mh, 1 - y);
+    result = result.then(s => s.add(
+      newStruct('', {x, y, w, h, name: v.name})));
```

Comment:
What optional fields?

Is it still a lot of boilerplate now that you fixed makeStructType?